lib/src/kvs/tests/timestamp_to_versionstamp.rs

```diff
@@ -17,26 +17,32 @@ async fn timestamp_to_versionstamp() {
 	// Give the current versionstamp a timestamp of 0
 	let mut tx = ds.transaction(true, false).await.unwrap();
 	tx.set_timestamp_for_versionstamp(0, "myns", "mydb", true).await.unwrap();
+	tx.complete_changes(false).await.unwrap();
 	tx.commit().await.unwrap();
 	// Get the versionstamp for timestamp 0
 	let mut tx = ds.transaction(true, false).await.unwrap();
 	let vs1 = tx.get_versionstamp_from_timestamp(0, "myns", "mydb", true).await.unwrap().unwrap();
+	tx.complete_changes(false).await.unwrap();
 	tx.commit().await.unwrap();
 	// Give the current versionstamp a timestamp of 1
 	let mut tx = ds.transaction(true, false).await.unwrap();
 	tx.set_timestamp_for_versionstamp(1, "myns", "mydb", true).await.unwrap();
+	tx.complete_changes(false).await.unwrap();
 	tx.commit().await.unwrap();
 	// Get the versionstamp for timestamp 1
 	let mut tx = ds.transaction(true, false).await.unwrap();
 	let vs2 = tx.get_versionstamp_from_timestamp(1, "myns", "mydb", true).await.unwrap().unwrap();
+	tx.complete_changes(false).await.unwrap();
 	tx.commit().await.unwrap();
 	// Give the current versionstamp a timestamp of 2
 	let mut tx = ds.transaction(true, false).await.unwrap();
 	tx.set_timestamp_for_versionstamp(2, "myns", "mydb", true).await.unwrap();
+	tx.complete_changes(false).await.unwrap();
 	tx.commit().await.unwrap();
 	// Get the versionstamp for timestamp 2
 	let mut tx = ds.transaction(true, false).await.unwrap();
 	let vs3 = tx.get_versionstamp_from_timestamp(2, "myns", "mydb", true).await.unwrap().unwrap();
+	tx.complete_changes(false).await.unwrap();
```

Comment:
@phughk This bug in the existing test was the cause of KeyAlreadyExists 🤦 

We persist the ID generator's state by calling complete_changes.
Missing complete_changes resulted in divergence between the ID generator and the namespace instances written using `tx.put`.

We run kv tests followed by API integration tests. If kv tests ran this broken `timestamp_to_versionstamp` test as the last one, all the subsequent API integration tests are affected by the divergence, getting duplicate IDs from the generator, resulting in `tx.put`ting duplicate namespace instances.

Reply:
I remember that @tobiemh previously suggested integrating `complete_changes` into `commit` so that we don't miss calling it 😿
We should definitely do so.